<?xml version="1.0"?>
<launch>
    <!-- Add all launch files that you want to run additionally to your robot bringup like moveit, etc... -->
    <arg name="robot" default="cob4-2"/>
    <arg name="robot_env" default="saturn-ingolstadt"/>
    <arg name="pkg_env_config" default="$(find msh_env_config)"/>
    <arg name="pkg_nav_config" default="$(find msh_navigation_config)"/>
    <arg name="pkg_nav_env_config" default="$(find msh_navigation_config)"/>
    <arg name="env-script" default="$(find cob_bringup)/env.sh"/>
    
    
    <arg name="int_res" default="0.5"/>
    <arg name="part_num" default="100"/>
    <arg name="ndt_match" default="false"/>
    <arg name="gain" default="3"/>
    <arg name="counter" default="1"/>
    <arg name="mapping" default="1.0"/>
    <arg name="stmgain" default="0.5"/>
    <arg name="optimize_pose" default="false"/>   

    <node name="bagfile" pkg="rosbag" type="play" args="--clock --start 38 /home/fmw-hb/bagfiles/filtered3_ekf.bag" output="screen"/>  
    
    <!--include file="$(find cob_bringup)/drivers/scan_unifier.launch">
			<arg name="robot" value="cob4-2"/>
    </include-->
    
    <include file="$(find msh_bringup)/launch/rviz.launch">
			<arg name="robot" value="cob4-2"/>
    </include>
    
      <arg name="map_server" default="true" />
      <arg name="grid_converter" default="false" />
      <arg name="localgrid" default="false" />
      <!-- sensor seetings -->
      <arg name="sensor_config" default="$(arg pkg_nav_env_config)/envs/$(arg robot_env)/nav/sensor_setup.yaml" />
      <arg name="slam" default="false"/>
      <arg name="imu_driver" default="false"/>
      <arg name="rfid" default="false"/>
      <arg name="unified_karto_laser_scan" default="true"/>
      <arg name="filter_karto_laser_scan" default="false"/>
      <arg name="license_package" default=""/>
      <!-- odometry settings -->
      <arg name="dead_reck" default="false" />

     <!--include file="$(find ipa_navigation_bringup)/localization/ipa_localization.xml" >
        <arg name="robot" value="cob4-2"/>
        <arg name="robot_env" value="saturn-ingolstadt" />
        <arg name="pkg_env_config" value="$(find msh_env_config)"/>
        <arg name="pkg_nav_config" value="$(find msh_navigation_config)"/>
        <arg name="pkg_nav_env_config" value="$(find msh_navigation_config)"/>
        <arg name="map" value="$(find msh_env_config)/envs/saturn-ingolstadt/map.yaml" />
        <arg name="slam" value="false" />
        <arg name="imu_driver" value="$(arg imu_driver)" />
        <arg name="rfid" value="$(arg rfid)" />
        <arg name="map_server" value="$(arg map_server)" />
        <arg name="grid_converter" value="$(arg grid_converter)" />
        <arg name="sensor_config" value="$(arg sensor_config)" />
        <arg name="dead_reck" value="$(arg dead_reck)" />
        <arg name="localgrid" value="$(arg localgrid)" />
     </include-->
        
    <!-- costmap zone filter -->
   <node name="laserscan_zone_filter" pkg="ipa_laserscan_zone_filter" type="laserscan_zone_filter_node" output="screen">
        <rosparam file="$(find msh_navigation_config)/robots/cob4-2/nav/laserscan_zone_filter.yaml" command="load" />
        <remap from="scan" to="scan_unified"/> 
        <remap from="scan_filtered" to="costmap_scan_filtered" />
    </node>

    <!-- hannes SLAM -->       
    <!--include file="$(find ipa_long_term_slam)/launch/long_term_slam.launch">
        <arg name="init_from_backup" value="false"/>
        <arg name="init_from_static_map" value="true"/>
        <arg name="robot" value="cob4-2"/>
        <arg name="robot_env" value="saturn-ingolstadt" />
        <arg name="pkg_env_config" value="$(find msh_env_config)"/>
        <arg name="map" value="$(find msh_env_config)/envs/saturn-ingolstadt/map.yaml" />
        <arg name="resolution" value="$(arg int_res)" />
        <arg name="particles" value="$(arg part_num)" />
        <arg name="weight_gain" value="$(arg gain)" />
    </include--> 
    
    <!-- ipa SLAM -->
    <include file="$(find ipa_long_term_slam)/launch/long_term_slam.launch">
        <arg name="init_from_backup" value="false"/>
        <arg name="init_from_static_map" value="true"/>
        <arg name="robot" value="cob4-2"/>
        <arg name="robot_env" value="saturn-ingolstadt" />
        <arg name="pkg_env_config" value="$(find msh_env_config)"/>
        <arg name="map" value="$(find msh_env_config)/envs/saturn-ingolstadt/map.yaml" />
    </include> 
   
</launch>
